winter/http/request_body_resolver.py

```diff
@@ -0,0 +1,92 @@
+import typing
+
+import dataclasses
+import pydantic
+from rest_framework.request import Request
+
+from .request_body import RequestBodyAnnotation
+from .request_body_exception_handler import InvalidRequestBodyException
+from .. import ArgumentResolver
+from .. import type_utils
+from ..core import ComponentMethodArgument
+from ..type_utils import is_iterable
+
+
+class InputDataArgumentResolver(ArgumentResolver):
+
+    def __init__(self):
+        super().__init__()
+        self._pydantic_dataclasses = {}
+
+    def is_supported(self, argument: ComponentMethodArgument) -> bool:
+        return argument.method.annotations.get_one_or_none(RequestBodyAnnotation) is not None
+
+    def resolve_argument(self, argument: ComponentMethodArgument, http_request: Request):
+        fields = dataclasses.fields(argument.type_)
+
+        input_data, missing_fields = self._get_input_data(fields, http_request)
+        errors = {}
+
+        pydantic_dataclass = self._get_pydantic_dataclass(argument.type_)
+
+        validated_data = {}
+
+        try:
+            validated_data = dataclasses.asdict(pydantic_dataclass(**input_data))
+        except pydantic.error_wrappers.ValidationError as exception:
+            self._update_errors(errors, exception.errors(), missing_fields)
```

Comment:
Do not pass `errors` to `_update_errors`, but return it. And rename the method to something like `_convert_pydantic_errors`